src/org/benf/cfr/reader/bytecode/analysis/parse/expression/CastExpression.java

```diff
@@ -57,6 +63,34 @@ public Expression deepClone(CloneHelper cloneHelper) {
         return new CastExpression(getLoc(), getInferredJavaType(), cloneHelper.replaceOrClone(child), forced);
     }
 
+    @Override
+    public Literal getComputedLiteral(Map<LValue, Literal> display) {
+        if (getChild() instanceof Literal) {
+            Literal childLiteral = (Literal) getChild();
+            CastAction action = childLiteral.getValue().getInferredJavaType().chain(getInferredJavaType());
```

Comment:
You don't need to chain here - type chaining (that code is very confusing) is used to generate type dependencies.

Reply:
Ah, I'm using this because I assumed this would allow to check if a cast is redundant easily _(by checking against none)_